delb/transform.py

```diff
@@ -0,0 +1,172 @@
+"""
+This module offers a canonical interface with the aim to make re-use of transforming
+algorithms easier.
+
+Let's look at it with examples::
+
+   from delb.transform import Transformation
+
+
+   class ResolveCopyOf(Transformation):
+       def transform(self):
+           for node in self.root.css_select("*[copyOf]"):
+               source_id = node["copyOf"]
+               source_node = delb.document.xpath(
+                   f'//*[@xml:id="{source_id[1:]}"]'
```

Comment:
this throws an Exception in `collections.UserString#__getitem__`. What is the slice supposed to mean!?